### PR TITLE
Update uncrustify version to fix warnings on armhf

### PR DESCRIPTION
**Pull request by emersonknapp**
Connected to https://github.com/ros2/ros2/issues/721

Fixes these warnings https://ci.ros2.org/view/All/job/test_ci_linux-armhf/8/warnings23Result/package.575780676/

This patch mirrors just-opened https://github.com/uncrustify/uncrustify/pull/2308 - once that change is bundled into a release, we can update the version we pull and drop the patch.

Signed-off-by: Emerson Knapp <eknapp@amazon.com>

### CODE REVIEW

**Comment by emersonknapp** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7291/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3466/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5966/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7110/)

Full re-run after other PRs merged
* Linux-armhf [Build Status](https://ci.ros2.org/view/All/job/test_ci_linux-armhf/13/)

**Review by thomas-moulard — APPROVED** [review]

**Review by clalancette — APPROVED** [review]
The code looks fine to me.  I'm going to hold off merging this until the upstream patch gets merged, then we can merge this in.  Thanks.

**Comment by dirk-thomas** [comment]
The upstream PR has been merged. Instead of merging this PR I would suggest to rebase the vendor code to the current master of upstream instead.

**Comment by emersonknapp** [comment]
The vendor code logic is downloading a tarball of a specific release tag, currently, (https://github.com/ament/uncrustify_vendor/blob/master/CMakeLists.txt#L49) - are you sure we want to point it at master? My thinking was that we would wait until the next release, which will include this fix, and remove the patch then.

**Comment by dirk-thomas** [comment]
> are you sure we want to point it at master?

Yes, that would be the intention. If you like to use the latest changes landed upstream using a commit hash from upstream sounds better than cherry-picking the change into our vendor package.

During that process it will need to be made sure that all our tests pass with the updated version of `uncrustify`.

**Comment by emersonknapp** [comment]
OK - thanks for confirming. I will update it

**Comment by emersonknapp** [comment]
Here is a run doing a full build and a `--mixin linters-only` test (with this PR and the two above Connected ones fixing formatting for latest uncrustify)

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7310/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3484/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/5987/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7130/)
* Linux-armhf [Build Status](https://ci.ros2.org/job/test_ci_linux-armhf/17/)
Note: the armhf build instability is only from remaining FastCDR warnings

**Comment by emersonknapp** [comment]
@clalancette do you think we're good to merge now? Updated to the upstream version after that was merged.

**Comment by clalancette** [comment]
@emersonknapp I'm happy with this PR, but in order not to break CI we'll need to land https://github.com/ros2/rcutils/pull/164 first/simultaneously.

**Comment by emersonknapp** [comment]
CI run against latest - only including this PR

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/7394/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/3551/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6068/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/7223/)
* Linux-armhf [Build Status](https://ci.ros2.org/job/ci_linux-armhf/13/)

**Comment by emersonknapp** [comment]
All upstream changes have been merged, the above armhf build only has the unrelated warnings that are being fixed in https://github.com/ros2/demos/pull/372

@clalancette good to merge?

**Comment by emersonknapp** [comment]
@clalancette ping

File: CMakeLists.txt
Comment (dirk-thomas) [inline CMakeLists.txt:50]: Keeping this comment unchanged is pretty confusing since the code doesn't use that specific version anymore.

**Pull request merged**
